kinto/core/permission/postgresql/migrations/migration_001_002.sql

```diff
@@ -0,0 +1,15 @@
+-- Same table as exists in the storage backend, but used to track
+-- migration status for both. Only one schema actually has to create
+-- it.
+CREATE TABLE IF NOT EXISTS metadata (
+    name VARCHAR(128) NOT NULL,
+    value VARCHAR(512) NOT NULL
+);
+
+ALTER TABLE user_principals
+    ALTER COLUMN user_id TYPE TEXT COLLATE "C";
+
+ALTER TABLE access_control_entries
+    ALTER COLUMN object_id TYPE TEXT COLLATE "C";
+
+INSERT INTO metadata VALUES ('permission_schema_version', '2');
```

Comment:
nit: If the `metadata` table already existed, it's not a guarantee that the column order is exactly `'name', 'value'`. A safer INSERT statement would be:
`INSERT INTO metadata (name, value) VALUES ('permission_schema_version', '2');`